src/server/system_services/bucket_server.js

```diff
@@ -201,6 +202,13 @@ async function create_bucket(req) {
     return get_bucket_info({ bucket: created_bucket });
 }
 
+async function validate_pool_constraints() {
+    if (config.ALLOW_BUCKET_CREATE_ON_INTERNAL !== true) {
```

Comment:
Sounds harsh for every create bucket, maybe calculate first time and as long as non optimal. 
Once optimal returned a "cached" reply of true ?

Reply:
It is possible, but the risk is that we will have invalid pools and allow the creation of buckets.

Reply:
What's the scenario in which we will allow with an invalid pool ?

Reply:
In case we had an OPTIMAL pool and it got broken (never comes back to OPTIMAL).
We will create buckets that will never be attached to a valid pool (until someone creates one).

Reply:
that's the same as we had before with the infinite loop once the conditions was satisfied once.

Reply:
Yeah, I'm not saying that I've added something bad. I'm just saying that in theory, we have the same consequences as previously.